tests/tracking/test_tracking.py

```diff
@@ -131,12 +127,12 @@ def test_start_run_context_manager(tracking_uri_mock):
     assert finished_run.info.status == RunStatus.FINISHED
     # Launch a separate run that fails, verify the run status is FAILED and the run UUID is
     # different
-    second_run = start_run()
-    assert second_run.info.run_uuid != first_uuid
```

Comment:
Example of a pitfall caused by starting a run outside a with block - if the assertion here fails, we never unset `second_run` as the active run, so starting runs in other tests fails